### PR TITLE
Feature to allow or disallow saving empty field is added

### PR DESCRIPTION
**Pull request by sanbornsen**

### CODE REVIEW

**Comment by sanbornsen** [comment]
The issue is in planner - https://github.com/openshiftio/openshift.io/issues/2538 
I thought it would be better to have a feature like this in this component in here.

**Comment by joshuawilson** [comment]
Can you add a test for the new feature?

File: src/app/markdown/markdown.component.html [hunk 1]
```diff
@@ -53,7 +54,7 @@
   <button
     (click)="saveClick()"
     class="fl btn btn-primary pull-right action-btn btn-save"
-    [disabled]="saving"
+    [disabled]="saving || (!allowEmptySave && fieldEmpty)"
```
Comment (vikram-raj) [inline src/app/markdown/markdown.component.html]: Same conditions should check for the disabled class as well in line no. 58.

Comment (sanbornsen) [reply src/app/markdown/markdown.component.html]: We don't actually need a class. It's a button so `disabled` property should work by itself

**Comment by sanbornsen** [comment]
@joshuawilson added tests for the new feature.

**Pull request merged**